transport/http/dial.go

```diff
@@ -41,7 +43,12 @@ func NewClient(ctx context.Context, opts ...option.ClientOption) (*http.Client,
 	if settings.HTTPClient != nil {
 		return settings.HTTPClient, endpoint, nil
 	}
-	trans, err := newTransport(ctx, defaultBaseTransport(ctx, clientCertSource), settings)
+
+	dialTLSContext, targetEndpoint := s2autil.ConfigureHTTPTransport(clientCertSource, endpoint, settings)
```

Comment:
so if you update the helper to return the original endpoint when dialTLSContext is nil, you can assign to "endpoint" here directly.

On a related note, we should try to consolidate the endpoint calculation logic between internal/dca.go and s2util for the sake of consistency. If you take a look at https://github.com/googleapis/google-api-go-client/blob/main/internal/dca.go#L98 , the endpoint calculation is non trivial and factors in many things such "IS MTLS env var always/auto/never" and whether a certificateSource is present. s2a should work the same way and honor user requests to disable mTLS for instance (this impacts internal tests). I recommend you rename internal/dca.go to internal/cba.go (Certificate Based Access), add new helpers that to file such as internal.GetDialTLSContextAndEndpoint so you can reuse the endpoint calculation logic. If you consider s2a mTLS equivalent to device certs, then the logic should be parallel. WDYT?

Reply:
thanks I made some code consolidation  with dca.go per your suggestion.